### PR TITLE
Update PAN Decoder support encoder depth

### PR DESCRIPTION
**Pull request by brianhou0208**
Hi @qubvel,

This PR adds support for `encoder_depth` in the PAN decoder while ensuring compatibility with previously trained models, allowing them to be loaded without any issues.

## Update
1. Added type annotations for improved code clarity and robustness.
2. Introduced support for encoder_depth with validation to handle various encoder configurations.
3. Reduced the size of tests/test_models.py while maintaining test coverage.

### CODE REVIEW

File: segmentation_models_pytorch/decoders/pan/model.py [hunk 1]
```diff
@@ -52,12 +56,13 @@ class PAN(SegmentationModel):
...
         encoder_name: str = "resnet34",
+        encoder_depth: Literal[3, 4, 5] = 5,
         encoder_weights: Optional[str] = "imagenet",
-        encoder_output_stride: int = 16,
+        encoder_output_stride: Literal[16, 32] = 16,
         decoder_channels: int = 32,
         in_channels: int = 3,
         classes: int = 1,
-        activation: Optional[Union[str, callable]] = None,
+        activation: Optional[Union[str, Callable]] = None,
```
Comment (qubvel) [inline segmentation_models_pytorch/decoders/pan/model.py:65]: Thanks for fixing type hints! ❤️

**Review by qubvel — COMMENTED** [review]
@brianhou0208 Thanks for the update!

File: segmentation_models_pytorch/decoders/pan/decoder.py [hunk 2]
```diff
@@ -175,34 +180,43 @@ def forward(self, x, y):
...
+        for i in range(1, len(encoder_channels)):
+            self.add_module(
+                f"gau{len(encoder_channels)-i}",
+                GAUBlock(
+                    in_channels=encoder_channels[i],
+                    out_channels=decoder_channels,
+                    upscale_mode=upscale_mode,
+                ),
+            )
+
```
[the remark this answers is not in the document]
Comment (brianhou0208) [reply segmentation_models_pytorch/decoders/pan/decoder.py]: I have updated the it

**Review by qubvel — APPROVED** [review]

**Comment by qubvel** [comment]
Thanks for your contribution!

**Pull request merged**